fabric-lifecycle-events-v1/src/main/java/net/fabricmc/fabric/mixin/event/lifecycle/WorldMixin.java

```diff
@@ -77,8 +79,15 @@ protected boolean onPurgeRemovedBlockEntities(List<BlockEntity> blockEntityList,
 			}
 		}
 
-		// Mimic vanilla logic
-		return blockEntityList.removeAll(removals);
+		// Mimic vanilla logic - with some perf boosts
+		// Gate perf boost behind a minimum amount of removals
+		if (removals.size() <= 4) {
```

Comment:
Why exactly 4? In vanilla survival, I guess that number won't appear at all. A useless check in essence.

Reply:
The number itself could be tweaked, allocating a new set is more costly than removing from List -> List with certain amounts of elements.

Reply:
We are talking about bases with maybe 1000 BEs in some chunks, so yes definitely needed.